timesketch/api/v1/resources_test.py

```diff
@@ -555,10 +757,112 @@ def test_get_sigma_rule(self):
+        self.assertEqual(response.status_code, HTTP_STATUS_CODE_BAD_REQUEST)
+
+    def test_get_rule_by_text_no_form_data(self):
+        """Authenticated request to get an sigma rule by text with no form
+        data"""
+        self.login()
+        data = dict(action="post")
+        response = self.client.post(
+            "/api/v1/sigmarule/text/",
+            data=json.dumps(data, ensure_ascii=False),
+            content_type="application/json",
+        )
+        data = json.loads(response.get_data(as_text=True))
+
+        self.assertIn("Missing value in the request", data["message"])
+        self.assertEqual(response.status_code, HTTP_STATUS_CODE_BAD_REQUEST)
+
+
+class SigmaByTextResourceTest(BaseTest):
+    """DEPRECATED: Test Sigma by text resource."""
+
+    correct_rule = """
+        title: Installation of foobar
+        id: bb1e0d1d-cd13-4b65-bf7e-69b4e740266b
+        description: Detects suspicious installation of foobar
+        references:
+            - https://samle.com/foobar
+        author: Alexander Jaeger
+        date: 2020/12/10
+        modified: 2020/12/10
+        tags:
+            - attack.discovery
+            - attack.t1046
+        logsource:
+            product: linux
+            service: shell
+        detection:
+            keywords:
+                # Generic suspicious commands
+                - '*apt-get install foobar*'
+            condition: keywords
+        falsepositives:
+            - Unknown
+        level: high
+        """
+    expected_response = {
+        "meta": {},
+        "objects": [
+            {
+                "title": "Installation of foobar",
+                "id": "bb1e0d1d-cd13-4b65-bf7e-69b4e740266b",
+                "description": "Detects suspicious installation of foobar",
+                "references": ["https://samle.com/foobar"],
+                "author": "Alexander Jaeger",
+                "date": "2020/12/10",
+                "modified": "2020/12/10",
+                "tags": ["attack.discovery", "attack.t1046"],
+                "logsource": {"product": "linux", "service": "shell"},
+                "detection": {
+                    "keywords": ["*apt-get install foobar*"],
+                    "condition": "keywords",
+                },
+                "falsepositives": ["Unknown"],
+                "level": "high",
+                "es_query": '(data_type:("shell:zsh:history" OR "bash:history:command" OR "apt:history:line" OR "selinux:line") AND "apt-get install foobar")',  # pylint: disable=line-too-long
+                "file_name": "N/A",
+                "file_relpath": "N/A",
+            }
+        ],
+    }
+
+    def test_get_sigma_rule(self):
+        """DEPRECATED: Authenticated request to get an sigma rule by text."""
+        self.login()
+
+        data = dict(content=self.correct_rule)
+        response = self.client.post(
+            "/api/v1/sigma/text/",
+            data=json.dumps(data, ensure_ascii=False),
```

Comment:
Do we really need `ensure_ascii` here? (and in other places?) given that the test strings we're using don't have non-ASCII characters in them.

Reply:
But here you have control over the text - so you know these characters are not gonna happen.